app/core/ai_cache.py

```
@@ -0,0 +1,106 @@
+from typing import Any, Optional, Union
```

Ruff (0.8.2)

`typing.Union` imported but unused

Remove unused import: `typing.Union`

(F401)